### PR TITLE
Updating README JIRA link to ASF JIRA

### PR DESCRIPTION
**Pull request by absurdfarce**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -80,7 +80,7 @@ See the [Cassandra error handling done right blog](https://www.datastax.com/blog
 * [FAQ]
 
 [API docs]: https://docs.datastax.com/en/drivers/java/4.17
-[JIRA]: https://datastax-oss.atlassian.net/browse/JAVA
+[JIRA]: https://issues.apache.org/jira/projects/CASSANDRA/issues/
```
Comment (absurdfarce) [inline README.md]: Mentioned in separate conversation by @ekaterinadimitrova2: we should mention the "Client/java-driver" component we've been using for Java driver tickets here.

Comment (absurdfarce) [reply README.md]: I'm kinda wondering if there isn't some way we could pre-fill in that component by customizing this link a bit, maybe with some query params or extra URL args...

Comment (michaelsembwever) [reply README.md]:
```suggestion
[JIRA]: https://issues.apache.org/jira/issues/?jql=project%20%3D%20CASSANDRA%20AND%20component%20%3D%20%22Client%2Fjava-driver%22%20ORDER%20BY%20key%20DESC
```
? 

there's no way to assign the component when creating a new issue by url or template…

Comment (absurdfarce) [reply README.md]: Ah, that's an excellent idea @michaelsembwever !  I'll also add some quick text here to make it clear that "Client/java-driver" is selected.

**Comment by absurdfarce** [comment]
Given the last round of changes I'm a +1 to merging this; the sooner we can get these instructions out to our users the better!

**Review by adutra — APPROVED** [review]

**Comment by absurdfarce** [comment]
I apparently did something very unpleasant to my local feature branch, upshot being it was easier to re-apply changes and start over.  Resulting was #1926 which has now been merged.

**Pull request closed without merging**